particles/test/transformations/test-multiplex-slots-particle.js

```diff
@@ -0,0 +1,75 @@
+// @license
+// Copyright (c) 2017 Google Inc. All rights reserved.
+// This code may only be used under the BSD style license found at
+// http://polymer.github.io/LICENSE.txt
+// Code distributed by Google as part of this project is also
+// subject to an additional IP rights grant found at
+// http://polymer.github.io/PATENTS.txt
+
+defineParticle(({DomParticle}) => {
+  return class MultiplexSlotsParticle extends DomParticle {
+    async setViews(views) {
+      let arc = await this.constructInnerArc();
+      var foosView = views.get('foos');
+      var foosList = await foosView.toList();
+      for (let [index, foo] of foosList.entries()) {
+        let fooView = await arc.createView(foosView.type.primitiveType(), "foo" + index);
+        let hostedParticle = await views.get('particle').get();
+        let hostedSlotName = [...hostedParticle.slots.values()][0].name;
+        let slotName = [...this.spec.slots.values()][0].name;
+        let slotId = await arc.createSlot(this, slotName, hostedParticle.name, hostedSlotName);
+
+        // TODO: how can i require assert here?
```

Comment:
assert needs to be passed into the environment which is provided to the defineParticle call, and you'd bind it next to DomParticle.

Reply:
probably better to do it in a separate change then.